File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Core/Bntx/BinaryData/BinaryConverterCache.cs
using System;
using System.Collections.Generic;

namespace MiniToolbox.Core.Bntx;

internal static class BinaryConverterCache
{
	private static readonly Dictionary<Type, IBinaryConverter> _cache = new Dictionary<Type, IBinaryConverter>();

	internal static IBinaryConverter GetConverter(Type type)
	{
		if (!_cache.TryGetValue(type, out var value))
		{
			value = (IBinaryConverter)Activator.CreateInstance(type);
			_cache.Add(type, value);
		}
		return value;
	}
}

File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Api/src/bntx/BinaryData/BinaryConverterCache.ts
import type { IBinaryConverter } from './IBinaryConverter.js';

/**
 * 1:1 port of BinaryConverterCache.cs
 * Caches IBinaryConverter instances by their constructor type.
 */
export class BinaryConverterCache {
    private static _cache = new Map<new () => IBinaryConverter, IBinaryConverter>();

    static getConverter(type: new () => IBinaryConverter): IBinaryConverter {
        let converter = this._cache.get(type);
        if (!converter) {
            converter = new type();
            this._cache.set(type, converter);
        }
        return converter;
    }
}
